keps/sig-release/20190316-rebase-images-to-distroless.md

```diff
@@ -0,0 +1,149 @@
+There are 4 main efforts on this KEP.
+
+1. The images are built via bazel. In such case, we will update the bazel BUILD rule to switch to base from distroless image. This method applies for the core containers like kube-apiserver, kube-controller-manager, cloud-controller-manager, kube-scheduler. As for kube-proxy, it is based on debian-iptable which distroless doesn't support iptables yet. Thus, kube-proxy won't be changed.
+2. The images have dependencies that are not supported by distroless. One typical example is the usage of shell. Previously, shell is widely used for redirecting glog output to a certain directory. This use case is no longer needed since we've switched from glog to klog which can accept a flag to specify the log output path. A generic approach is: Remove the dependencies that distroless doesn't support and then rebase the images to distroless (e.g. [issue/1787](https://github.com/kubernetes/autoscaler/issues/1787). Meanwhile, we limit the introduction of new dependencies. (e.g. [pr/74690](https://github.com/kubernetes/kubernetes/pull/74690#discussion_r266037189))
+3. Images based "FROM scratch" is safe to switch to distroless directly.
+4. Images from kubernetes incubator won't be changed directly by this KEP and the release plan is not estimated here. We notify the project OWNERs and we defer to the OWNERs on whether or not those images should be updated.
+
+## Graduation Criteria
+
+This KEP is targeted at v1.15 release. The full list of images switched to distroless will be updated later on.
+
+### Testing
+
+#### Test the Core Images.
+
+Step 1: Use bazel to build the component .tar files.
+```bash
+>> make bazel-release
+```
+output .tar files
+- bazel-bin/build/release-tars/kubernetes-client-linux-amd64.tar.gz
+- bazel-bin/build/release-tars/kubernetes-node-linux-amd64.tar.gz
+- bazel-bin/build/release-tars/kubernetes-server-linux-amd64.tar.gz
+- bazel-bin/build/release-tars/kubernetes-manifests.tar.gz
+- bazel-bin/build/release-tars/kubernetes-test-portable.tar.gz
+- bazel-bin/build/release-tars/kubernetes-test-linux-amd64.tar.gz
+- bazel-bin/build/release-tars/kubernetes-test.tar.gz
+- bazel-bin/build/release-tars/kubernetes.tar.gz
+- bazel-bin/build/release-tars/kubernetes-src.tar.gz
+Step 2: Upload .tar to the bigstore (Google GKE internal test env)
+Step 3: Create a cluster using the uploaded .tar (Google GKE internal test env)
+```bash
+>> gcloud container clusters create --cluster-version 1.14.0-yuwen-distroless-0308115 distroless-test-cluster
```

Comment:
These internal GKE details aren't relevant to the opensource KEP. We have opensource test setups that are more relevant. You can also say that we will test on GKE setups, and offer preview versions if other distributors wish to test the images on their platforms.

Reply:
The opensource test doesn't go through the release process (it uses a different test config which based on gcr.k8s.io/pause). After poking around and chatting with multiple people, the internal GKE  is the only workable solution I found (and it's handy). 

I agree with you that I shouldn't put too much GKE specific info in this KEP. I'm not sure how much value this test section gives, I can skip all the details and give a brief description. Thoughts?